## Гипотеза: Если наряду с фото товара в карточках использовать видеоролики, конверсия просмотров в продажи вырастет с 2.1% до 2.6%, т.к. по статистике видеоконтент привлекает внимание покупателей больше, чем фото.

### Что делаем
### *Контрольная версия: показываем только фото товара в карточках (оставляем, как есть сейчас). Тестовая версия: показываем не только фото товара, но и короткие видеоролики о товаре в карточках.*

### На каких пользователях тестируем
### *Все пользователи, которые заходят на страницу с товаром.*

### Метрики
### *Конверсия просмотров страницы с товаром в продажи.*

### План действий
### *Если конверсия просмотров в продажи действительно увеличится с текущих 2.1% до предполагаемых 2.6%, во все карточки товаров добавим видеоролики о товаре для всех потенциальных покупателей.*
### *Если предполагаемую конверсию достичь не удастся, оставляем в карточках только фото.*

In [2]:
from statsmodels.stats.power import tt_ind_solve_power, zt_ind_solve_power
from statsmodels.stats.proportion import proportion_effectsize
from statsmodels.stats.meta_analysis import effectsize_smd
from typing import Union
import plotly.graph_objects as go
from scipy import stats
from math import asin
import numpy as np

#### Для определения продолжительности проведения эксперимента посчитаем необходимое количество наблюдений. Нам известны alpha = 5%, мощность эксперимента = 1-beta = 1-0.2 = 0.8 = 80%, посчитаем effect size с помощью z-критерия для пропорций, т.к. целевой метрикой является конверсия:

In [9]:
def calc_proportion_effect_size(prob_1: float, prob_2: float):
    return abs(proportion_effectsize(prob_1, prob_2))

In [10]:
conv_1, conv_2 = 0.021, 0.026
es_prop = calc_proportion_effect_size(conv_1, conv_2)
es_prop

0.033052736898224966

#### Подставим значение effect size в функцию, пятым неизвестным элементом которой является nobs - количество наблюдений:

In [13]:
def calc_sample_size_proportion(effect_size: float,
                                alpha: float = .05,
                                beta: float = .2,
                                ratio: Union[float, int] = 1):
    
    n = zt_ind_solve_power(effect_size=effect_size,
                           alpha=alpha,
                           power=(1 - beta),
                           ratio=ratio,
                  )
    return int(n)

In [14]:
calc_sample_size_proportion(es_prop)

14368

#### 14368 наблюдений для каждой (контрольной и тестовой) версии потребуется для завершения эксперимента и фиксации результата/выводов. Если предположить, что ежедневное количество просмотров страницы карточки товара ~2000, нам потребуется (14368*2)/2000=14.37 дней.